pkg/deploy/controller/deploymentconfig/controller_test.go

```diff
@@ -1,835 +1,629 @@
+				{version: 4, replicas: 1, replicasA: newint(1), status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 5, replicas: 0, replicasA: newint(0), desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: false},
+			},
+			errExpected: false,
+		},
+		{
+			name:             "(compat) already deployed, no active deployment",
+			replicas:         2,
+			newVersion:       2,
+			expectedReplicas: 1,
+			before: []deployment{
+				{version: 1, replicas: 0, desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: false},
+				{version: 2, replicas: 0, desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: false},
+			},
+			after: []deployment{
+				{version: 1, replicas: 0, replicasA: newint(0), desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: false},
+				{version: 2, replicas: 0, replicasA: newint(0), desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: false},
+			},
+			errExpected: false,
+		},
+		{
+			name:             "(compat) scale up latest/active completed deployment",
+			replicas:         1,
+			newVersion:       2,
+			expectedReplicas: 5,
+			before: []deployment{
+				{version: 1, replicas: 0, status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 5, status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			after: []deployment{
+				{version: 1, replicas: 0, replicasA: newint(0), status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 5, replicasA: newint(5), status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			errExpected: false,
+		},
+		{
+			name:             "(compat) scale up active (not latest) completed deployment (RC targetted directly)",
+			replicas:         1,
+			newVersion:       2,
+			expectedReplicas: 5,
+			before: []deployment{
+				{version: 1, replicas: 5, status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 0, desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: true},
+			},
+			after: []deployment{
+				{version: 1, replicas: 5, replicasA: newint(5), status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 0, replicasA: newint(0), desiredA: newint(1), status: deployapi.DeploymentStatusFailed, cancelled: true},
+			},
+			errExpected: false,
+		},
+		/*
+			// TODO: In this case, I'm not sure how we can detect that the client has
+			// manually scaled the latest RC directly: the replica count might be less
+			// than desired if the deployment partially succeeded during a failure or
+			// cancellation, and it might be greater than desired due to a strategy
+			// bug (although the latter case seems like a more likely candidate for an
+			// assumption, which could maybe fix scale up most of the time but still
+			// leaves scale down broken.)
```

Comment:
@smarterclayton 

This is one of two test cases demonstrating a compatibility gap. When the active (but not latest) deployment has a replica count which differs from the last observed, we don't know for sure it was due to an external scaling event and not a partial (or buggy) rollout from the failed deployment.